gprofiler/profilers/profiler_base.py

```diff
@@ -239,6 +238,7 @@ def start(self) -> None:
         super().start()
 
         if self._profile_spawned_processes:
+            logger.debug(f"{self.__class__.__name__}: starting profiling spawning processes")
```

Comment:
Maybe the log message wasn't good enough, but it was in `_start_profiling_spawning` on purpose. I use this log message in the test to make sure that the profiling session started and that the profiler actually started detecting spawning processes. If the log is here then we can't know for sure that it won't catch the process as a preexisting process for the session.

Reply:
Hmm. It was confusing on that function because it kept happening forever and it was printed also when profiling spawned processes was disabled.

Maybe can be sovled with mocking? (or we run the container here...)

Reply:
I have another idea, pushing a patch

Reply:
Fixed it @ty93